Tests/Amqp10Tests.cs

```diff
@@ -374,6 +371,56 @@ public void ValidateMessagesFromGo()
             Assert.Equal("amqpValue", msgHeader.AmqpValue);
         }
 
+        [Fact]
+        public void ValidateMapsType()
+        {
+            const double DoubleValue = 6665555.34566;
+            var dt = DateTime.Now;
+            var app = new ApplicationProperties()
+            {
+                ["double_value"] = DoubleValue,
+                ["string_value"] = "test",
+                ["bool_value"] = true,
+                ["byte_value"] = (byte)1,
+                ["short_value"] = (short)1,
+                ["int_value"] = 1,
+                ["long_value"] = 1L,
+                ["ulong_value"] = 1UL,
+                ["float_value"] = 1.0f,
+                ["date_value"] = dt
+            };
+            var m = new Message(Encoding.Default.GetBytes("hello")) { ApplicationProperties = app };
+
+            Assert.NotNull(m.ApplicationProperties);
+
+            var size = DescribedFormatCode.Size;
+            size += sizeof(byte); //FormatCode.List32
+            size += sizeof(uint); // field numbers
+            size += sizeof(uint); // PropertySize
+            size += AmqpWireFormatting.GetAnySize("double_value");
+            size += AmqpWireFormatting.GetAnySize(DoubleValue);
+            size += AmqpWireFormatting.GetAnySize("string_value");
+            size += AmqpWireFormatting.GetAnySize("test");
+            size += AmqpWireFormatting.GetAnySize("bool_value");
+            size += AmqpWireFormatting.GetAnySize(true);
+            size += AmqpWireFormatting.GetAnySize("byte_value");
+            size += AmqpWireFormatting.GetAnySize((byte)1);
+            size += AmqpWireFormatting.GetAnySize("short_value");
+            size += AmqpWireFormatting.GetAnySize((short)1);
+            size += AmqpWireFormatting.GetAnySize("int_value");
+            size += AmqpWireFormatting.GetAnySize(1);
+            size += AmqpWireFormatting.GetAnySize("long_value");
+            size += AmqpWireFormatting.GetAnySize(1L);
+            size += AmqpWireFormatting.GetAnySize("ulong_value");
+            size += AmqpWireFormatting.GetAnySize(1UL);
+            size += AmqpWireFormatting.GetAnySize("float_value");
+            size += AmqpWireFormatting.GetAnySize(1.0f);
+            size += AmqpWireFormatting.GetAnySize("date_value");
+            size += AmqpWireFormatting.GetAnySize(dt);
+
+            Assert.Equal(size, m.ApplicationProperties.Size);
```

Comment:
I'm not very enthusiastic about this test case. The issue with aggregating all the sizes and asserting on the aggregate, is that you don't know if the individual size for each type is correct.

For example, I can "break" this test by swapping the value of `GetFloatSize()` and `GetDoubleSize()`. The test is still green, but the value returned by those two functions is be incorrect.

I think we must assert on each individual call to `GetAnySize()` to guarantee that we are returning the right thing.

Reply:
You are right [Done here](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/158/commits/32439232382b49f0127f30a4b42dc741699e6375)